### ISSUE TITLE
panic: total does not match supply

### ISSUE THREAD

**Issue by yillkid** [iotaledger/hornet#602]
**Describe the bug**
1. Get `Active: failed` status on systemd service

2. Get `panic: total does not match supply: 0 != 2779530283277761` journalctl log when HORNET recovery from power went out.

**To Reproduce**
- Cold shutdown the server
- Power on the server
- Get `panic: total does not match supply: 0 != 2779530283277761` error message

**Expected behavior**
Get `Active: active (running)` status on systemd service

**Environment information:**
 - OS: Ubuntu 18.04.4 LTS
 - RAM: 10 GB
 - Cores: 16 Cores
 - Type: Virtual Machine
 - Browser 
 - HORNET version: 0.4.2

**Additional context**
#### systemctl status:
```
● hornet.service - Hornet IOTA
   Loaded: loaded (/etc/systemd/system/hornet.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Mon 2020-08-10 01:28:21 UTC; 38min ago
  Process: 16259 ExecStart=/usr/bin/docker run --name hornet --net=host --user=39999 --cap-drop=ALL -v /etc/localtime:/etc/localtime:ro,Z -v /var/lib/hornet/mainnetdb:/app/mainnetdb:rw,Z -v 
  Process: 16232 ExecStartPre=/bin/bash -c /usr/bin/docker rm hornet || /bin/true (code=exited, status=0/SUCCESS)
 Main PID: 16259 (code=exited, status=2)

Aug 10 01:28:21 node8 docker[16259]: github.com/iotaledger/hive.go/node.(*Node).configure(0xc000420930, 0xc0000d09a0, 0x15, 0x16)
Aug 10 01:28:21 node8 docker[16259]:         /github/home/go/pkg/mod/github.com/iotaledger/hive.go@v0.0.0-20200623195435-4722a3081401/node/node.go:93 +0xf6
Aug 10 01:28:21 node8 docker[16259]: github.com/iotaledger/hive.go/node.New(0xc0002f1e98, 0x1, 0x1, 0x109de00)
Aug 10 01:28:21 node8 docker[16259]:         /github/home/go/pkg/mod/github.com/iotaledger/hive.go@v0.0.0-20200623195435-4722a3081401/node/node.go:38 +0x153
Aug 10 01:28:21 node8 docker[16259]: github.com/iotaledger/hive.go/node.Run(0xc0002f1e98, 0x1, 0x1, 0xc00026ca40)
Aug 10 01:28:21 node8 docker[16259]:         /github/home/go/pkg/mod/github.com/iotaledger/hive.go@v0.0.0-20200623195435-4722a3081401/node/node.go:51 +0x3f
Aug 10 01:28:21 node8 docker[16259]: main.main()
Aug 10 01:28:21 node8 docker[16259]:         /__w/hornet/hornet/main.go:67 +0x245
Aug 10 01:28:21 node8 systemd[1]: hornet.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Aug 10 01:28:21 node8 systemd[1]: hornet.service: Failed with result 'exit-code'.
```

#### journalctl:
```
Aug 10 01:28:19 node8 docker[16259]:               ██║  ██║██╔═══██╗██╔══██╗████╗  ██║██╔════╝╚══██╔══╝
Aug 10 01:28:19 node8 docker[16259]:               ███████║██║   ██║██████╔╝██╔██╗ ██║█████╗     ██║
Aug 10 01:28:19 node8 docker[16259]:               ██╔══██║██║   ██║██╔══██╗██║╚██╗██║██╔══╝     ██║
Aug 10 01:28:19 node8 docker[16259]:               ██║  ██║╚██████╔╝██║  ██║██║ ╚████║███████╗   ██║
Aug 10 01:28:19 node8 docker[16259]:               ╚═╝  ╚═╝ ╚═════╝ ╚═╝  ╚═╝╚═╝  ╚═══╝╚══════╝   ╚═╝
Aug 10 01:28:19 node8 docker[16259]:                                    v0.4.2
Aug 10 01:28:19 node8 docker[16259]: 2020-08-10T01:28:19Z        INFO        CLI        Profile mode 'auto', Using profile '8gb'
Aug 10 01:28:19 node8 docker[16259]: 2020-08-10T01:28:19Z        INFO        CLI        Loading plugins ...
Aug 10 01:28:19 node8 docker[16259]: 2020-08-10T01:28:19Z        INFO        Node        Loading Plugin: CLI ... done
Aug 10 01:28:19 node8 docker[16259]: 2020-08-10T01:28:19Z        INFO        Node        Loading Plugin: Graceful Shutdown ... done
Aug 10 01:28:19 node8 docker[16259]: 2020-08-10T01:28:19Z        INFO        Node        Loading Plugin: Profiling ... done
Aug 10 01:28:19 node8 docker[16259]: 2020-08-10T01:28:19Z        INFO        Node        Loading Plugin: Database ... done
Aug 10 01:28:19 node8 docker[16259]: 2020-08-10T01:28:19Z        INFO        Local        Initialized local: peer://9ct1AKfmnvLfX6Xat8QXVsdWgGxED7AoVwW6f8wsJ1Tt@0.0.0.0:14626
Aug 10 01:28:21 node8 docker[16259]: 2020-08-10T01:28:20Z        INFO        Node        Loading Plugin: Autopeering ... done
Aug 10 01:28:21 node8 docker[16259]: 2020-08-10T01:28:21Z        INFO        Node        Loading Plugin: WebAPI ... done
Aug 10 01:28:21 node8 docker[16259]: 2020-08-10T01:28:21Z        INFO        Node        Loading Plugin: Gossip ... done
Aug 10 01:28:21 node8 docker[16259]: SnapshotInfo:
Aug 10 01:28:21 node8 docker[16259]:         CooAddr: EQSAUZXULTTYZCLNJNTXQTQHOMOFZERHTCGTXOLTVAHKSA9OGAZDEKECURBRIXIJWNPFCQIOVFVVXJVD9
Aug 10 01:28:21 node8 docker[16259]:         SnapshotIndex: 1507002 (OW9UU9JOEUHRULLPXWGOIORJW9LITCSHNNBZYLCN9LPKNQINHS9CBHSIMFILNVQUA9ZHYIOZDRTFZ9999)
Aug 10 01:28:21 node8 docker[16259]:         EntryPointIndex: 1050000
Aug 10 01:28:21 node8 docker[16259]:         PruningIndex: 1050000
Aug 10 01:28:21 node8 docker[16259]:         Timestamp: 2020-07-31 22:41:03 +0000 UTC
Aug 10 01:28:21 node8 docker[16259]:         SpentAddressesEnabled: true
Aug 10 01:28:21 node8 docker[16259]: 2020-08-10T01:28:21Z        INFO        Node        Loading Plugin: Tangle ... done
Aug 10 01:28:21 node8 docker[16259]: 2020-08-10T01:28:21Z        INFO        Node        Loading Plugin: Peering ... done
Aug 10 01:28:21 node8 docker[16259]: 2020-08-10T01:28:21Z        INFO        Node        Loading Plugin: WarpSync ... done
Aug 10 01:28:21 node8 docker[16259]: 2020-08-10T01:28:21Z        INFO        Node        Loading Plugin: Tip-Sel ... done
Aug 10 01:28:21 node8 docker[16259]: 2020-08-10T01:28:21Z        INFO        Node        Loading Plugin: Metrics ... done
Aug 10 01:28:21 node8 docker[16259]: panic: total does not match supply: 0 != 2779530283277761
Aug 10 01:28:21 node8 docker[16259]: goroutine 1 [running]:
Aug 10 01:28:21 node8 docker[16259]: github.com/gohornet/hornet/pkg/model/tangle.GetLedgerStateForLSMIWithoutLocking(0x0, 0xc00067cb00, 0xf3, 0xf3, 0xc000794640)
Aug 10 01:28:21 node8 docker[16259]:         /__w/hornet/hornet/pkg/model/tangle/ledger_db.go:364 +0x34e
Aug 10 01:28:21 node8 docker[16259]: github.com/gohornet/hornet/pkg/model/tangle.GetLedgerStateForLSMI(0x0, 0x0, 0x0, 0x0, 0x0)
Aug 10 01:28:21 node8 docker[16259]:         /__w/hornet/hornet/pkg/model/tangle/ledger_db.go:376 +0x82
Aug 10 01:28:21 node8 docker[16259]: github.com/gohornet/hornet/plugins/snapshot.configure(0xc000692cc0)
Aug 10 01:28:21 node8 docker[16259]:         /__w/hornet/hornet/plugins/snapshot/plugin.go:112 +0xb12
Aug 10 01:28:21 node8 docker[16259]: github.com/iotaledger/hive.go/node.pluginCaller(0xfedda0, 0x20064b0, 0xc0002dace0, 0x1, 0x1)
Aug 10 01:28:21 node8 docker[16259]:         /github/home/go/pkg/mod/github.com/iotaledger/hive.go@v0.0.0-20200623195435-4722a3081401/node/events.go:14 +0x62
Aug 10 01:28:21 node8 docker[16259]: github.com/iotaledger/hive.go/events.(*Event).Trigger(0xc0006915c0, 0xc0002dace0, 0x1, 0x1)
Aug 10 01:28:21 node8 docker[16259]:         /github/home/go/pkg/mod/github.com/iotaledger/hive.go@v0.0.0-20200623195435-4722a3081401/events/event.go:32 +0x10d
Aug 10 01:28:21 node8 docker[16259]: github.com/iotaledger/hive.go/node.(*Node).configure(0xc000420930, 0xc0000d09a0, 0x15, 0x16)
Aug 10 01:28:21 node8 docker[16259]:         /github/home/go/pkg/mod/github.com/iotaledger/hive.go@v0.0.0-20200623195435-4722a3081401/node/node.go:93 +0xf6
Aug 10 01:28:21 node8 docker[16259]: github.com/iotaledger/hive.go/node.New(0xc0002f1e98, 0x1, 0x1, 0x109de00)
Aug 10 01:28:21 node8 docker[16259]:         /github/home/go/pkg/mod/github.com/iotaledger/hive.go@v0.0.0-20200623195435-4722a3081401/node/node.go:38 +0x153
Aug 10 01:28:21 node8 docker[16259]: github.com/iotaledger/hive.go/node.Run(0xc0002f1e98, 0x1, 0x1, 0xc00026ca40)
Aug 10 01:28:21 node8 docker[16259]:         /github/home/go/pkg/mod/github.com/iotaledger/hive.go@v0.0.0-20200623195435-4722a3081401/node/node.go:51 +0x3f
Aug 10 01:28:21 node8 docker[16259]: main.main()
Aug 10 01:28:21 node8 docker[16259]:         /__w/hornet/hornet/main.go:67 +0x245
Aug 10 01:28:21 node8 systemd[1]: hornet.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Aug 10 01:28:21 node8 systemd[1]: hornet.service: Failed with result 'exit-code'.
```

**Comment by yillkid** [comment]
Download and replace the [local snapshot file (latest-export.bin) ](https://dbfiles.iota.org/mainnet/hornet/latest-export.bin), still problem.

**Comment by ericvrp** [comment]
When hornet can’t cleanly shutdown there is a good chance the database gets corrupted. Redownloading the snapshot is a good start but that gets only done and used when there is no database available. So you should probably also delete your database. Looking at the logs you provided it looks like you database is at /var/lib/hornet/mainnetdb

Hornet-devs - please correct me if I’m wrong!

**Comment by muXxer** [comment]
@ericvrp you are correct. Revalidation of the database is fixed in #606
